src/ensembl/stories/shared-components/autosuggest-search-field/AutosuggestSearchField.stories.tsx

```diff
@@ -88,67 +91,80 @@ const Wrapper = (props: any) => {
-            selected will submit current content of the search field (see the
-            Actions panel).
-          </p>
-          <Wrapper
-            searchField={AutosuggestSearchField}
-            matchGroups={groupedMatches}
-            allowRawInputSubmission={true}
-            rightCorner={<RightCorner />}
-          />
-        </div>
-        <div className={styles.example}>
-          <p>
-            When no matches are available, pressing enter will submit the
-            current content of the field.
-          </p>
-          <Wrapper
-            searchField={AutosuggestSearchField}
-            allowRawInputSubmission={true}
-            rightCorner={<RightCorner />}
-          />
-        </div>
-      </div>
-    ),
-    { notes }
-  );
+export const MatchesSubmissionStory = (args: DefaultArgs) => (
+  <div className={styles.container}>
+    <p>For description of the component's behaviour, see the Notes tab.</p>
+    <div className={styles.example}>
+      <p>
+        Notice how the first match is immediately selected. Check the Actions
+        panel to see what gets submitted.
+      </p>
+      <Wrapper
+        searchField={AutosuggestSearchField}
+        matchGroups={groupedMatches}
+        rightCorner={<RightCorner />}
+      />
+    </div>
+    <div className={styles.example}>
+      <p>
+        If no matches were found, pressing Enter will not submit the raw search
+        string.
+      </p>
+      <Wrapper
+        searchField={AutosuggestSearchField}
+        rightCorner={<RightCorner />}
+        {...args}
+      />
+    </div>
+  </div>
+);
+
+MatchesSubmissionStory.story = {
+  name: 'allowing only submission of matches'
+};
+
+export const RawSearchSubmissionStory = (args: DefaultArgs) => (
+  <div className={styles.container}>
+    <div className={styles.example}>
+      <p>
+        Notice that, as opposed to the other variant, first suggestion is not
+        automatically pre-selected. Pressing enter when no suggestion is
+        selected will submit current content of the search field (see the
+        Actions panel).
+      </p>
+      <Wrapper
+        searchField={AutosuggestSearchField}
+        matchGroups={groupedMatches}
+        allowRawInputSubmission={true}
+        rightCorner={<RightCorner />}
+      />
+    </div>
+    <div className={styles.example}>
+      <p>
+        When no matches are available, pressing enter will submit the current
+        content of the field.
+      </p>
+      <Wrapper
+        searchField={AutosuggestSearchField}
+        allowRawInputSubmission={true}
+        rightCorner={<RightCorner />}
+        {...args}
+      />
+    </div>
+  </div>
+);
+
+RawSearchSubmissionStory.story = {
+  name: 'allowing raw search submission'
+};
+
+export default {
+  title: 'Components/Shared Components/AutosuggestSearchField',
+  parameters: {
+    notes
+  },
+  argTypes: {
+    onSelect: { action: 'selected' },
+    onSubmit: { action: 'submitted' }
```

Comment:
I get JS error when I try to use this component. I guess these should be `onSelected` & `onSubmitted`.

Reply:
Thanks for spotting this! 👍 Should be fixed now.